can/interfaces/vector/canlib.py

```diff
@@ -0,0 +1,176 @@
+                    hw_channel=hw_channel)
+                if xl_status.value == vxlapi.XL_SUCCESS:
+                    LOG.debug('Channel index %d found', app_channel.value)
+                    hw_type = ctypes.c_int(hw_type.value)
+                    hw_index = ctypes.c_int(hw_index.value)
+                    hw_channel = ctypes.c_int(hw_channel.value)
+                    channel_mask.append(
+                        vxlapi.get_channel_mask(hw_type, hw_index, hw_channel))
+                    LOG.debug('Channel %d, Type: %d, Mask: %d',
+                              hw_channel.value, hw_type.value,
+                              channel_mask[app_channel.value].value)
+            # Open one port for all channels
+            self.mask = vxlapi.XLaccess(0)
+            for CM in channel_mask:
+                self.mask.value += CM.value
+            self.port_handle = vxlapi.XLportHandle(
+                vxlapi.XL_INVALID_PORTHANDLE)
+            access_mask = self.mask
+            permission_mask = self.mask
+            xl_status = vxlapi.open_port(
+                user_name=self._app_name,
+                port_handle=self.port_handle,
+                access_mask=access_mask,
+                permission_mask=permission_mask)
+            LOG.debug(
+                'Open Port: PortHandle: %d, PermissionMask: %s, Status: %d',
+                self.port_handle.value, permission_mask.value, xl_status.value)
+            self.set_filters(can_filters)
+        return xl_status
+
+    def _go_on_bus(self):
+        vxlapi.activate_channel(
+            port_handle=self.port_handle, access_mask=self.mask)
+
+    def set_filters(self, can_filters=None):
+        if can_filters:
+            # Only one filter per ID type allowed
+            if len(can_filters) == 1 or (
+                    len(can_filters) == 2 and
+                    can_filters[0].get("extended") != can_filters[1].get("extended")):
+                for can_filter in can_filters:
+                    try:
+                        vxlapi.can_set_channel_acceptance(
+                            self.port_handle, self.mask,
+                            can_filter["can_id"], can_filter["can_mask"],
+                            vxlapi.XL_CAN_EXT if can_filter.get("extended") else vxlapi.XL_CAN_STD)
+                    except vxlapi.VectorError as exc:
+                        LOG.warning("Could not set filters: %s", exc)
+            else:
+                LOG.warning("Only one filter per extended or standard ID allowed")
+
+    def recv(self, timeout=None):
+        end_time = time.time() + timeout if timeout is not None else None
+        event = vxlapi.XLevent(0)
+        while True:
+            event_count = ctypes.c_uint(1)
+            try:
+                vxlapi.receive(self.port_handle, event_count, event)
+            except vxlapi.VectorError as exc:
+                if exc.status != vxlapi.XL_ERR_QUEUE_IS_EMPTY:
+                    raise
```

Comment:
We don't want to log empty queue exception since that will happen hundreds of times per second.